### PR TITLE
feat(actionlint): fix the errors produced by the ActionLint tool

### PR DESCRIPTION
**Pull request by ruzell22**
### **Commit** to be reviewed
---
feat(actionlint): fix the errors produced by the ActionLint tool
```
Primary Changes
---------------

1. Fix errors found by Actionlint on multiple yaml files
2. Temporarily removed test_weaver*.yaml, weaver_deploy*.yml, weaver/ directory in ActionLint
```
fixes: #2651

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Comment by ruzell22** [comment]
Addressed the errors found by Actionlint excluding on test-weaver and weaver-deploy yaml files. Will create a separate issue for that and add this issue ticket as related.

**Review by outSH — APPROVED** [review]

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

File: .github/workflows/.dast-nuclei-cmd-api-server.yaml
Comment (ruzell22) [reply .github/workflows/.dast-nuclei-cmd-api-server.yaml:97]: This error is shellcheck error "[SC2094: Make sure not to read and write the same file in the same pipeline.](https://github.com/koalaman/shellcheck/wiki/SC2094)" It was suggested to write the output to a temp file and reading it to the actual file to fix the error.

File: .github/workflows/actionlint.yaml
Comment (ruzell22) [reply .github/workflows/actionlint.yaml:69]: This error is shellcheck error "[SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.](https://github.com/koalaman/shellcheck/wiki/SC2242)" And then I used `exit 1` because this actionlint summary says if != 0 and do echo only when actionlint action failed.

**Review by petermetz — CHANGES_REQUESTED** [review]
@ruzell22 Looks good, just one thing: Let's try to get the actionlint job to green by deleting the test-weaver*.yml files and the ./weaver/directory for now.

That way when the new issue you created specifically for the weaver components is resolved, we can re-enable the actionlint scanning on these files as well (and keep the check green)
https://github.com/hyperledger/cacti/issues/3243

**Review by petermetz — CHANGES_REQUESTED** [review]
@ruzell22 Sorry, what I meant is deleting those files within the context of the running job from the temporary file system of the GitHub Action server, not from version control altogether.

This is the relevant command you want to update or extend:

```yaml
   - name: wipe_non_yaml_sources
      run: rm -rf packages/ examples/ extensions/ package.json
```

**Review by jagpreetsinghsasan — COMMENTED** [review]
Looks good to me now, will wait for @petermetz approval as the final approval from our org.

**Review by petermetz — APPROVED** [review]
Thank you @jagpreetsinghsasan 

@ruzell22 Also thank you, LGTM, we are good to go!

**Pull request merged**
